backend/app/services/ingestion/RemoteIngestWorker.scala

```diff
@@ -2,33 +2,100 @@ package services.ingestion
+              else Left(RemoteIngestFailure(s"Remote ingest job ${parsedJob.id} did not complete successfully. \n$parsedJob"))
+            )
+            job <- remoteIngestStore.getRemoteIngestJob(parsedJob.id)
+            workspaceMetadata <- annotations.getWorkspaceMetadata(job.userEmail, job.workspaceId)
+            _ <- remoteIngestStore.updateRemoteIngestJobStatus(parsedJob.id, "INGESTING")
+            ingestion <- Collections.createIngestionIfNotExists(Uri(job.collection), job.ingestion, manifest, index, pages, s3Config)
+            ingestFileResult <- IngestStorePolling.fetchData(job.ingestionKey, ingestStorage, scratchSpace){(path, fingerprint) =>
+              val collectionUri = Uri(job.collection)
+              Await.result(
+                new IngestFile(
+                  collectionUri,
+                  ingestionUri = Uri(ingestion.uri),
+                  uploadId = job.id,
+                  workspace = Some(WorkspaceItemUploadContext(
+                    workspaceId = job.workspaceId,
+                    workspaceNodeId = UUID.randomUUID().toString,
+                    workspaceParentNodeId = job.parentFolderId,
+                    workspaceName = workspaceMetadata.name
+                  )),
+                  username = job.userEmail,
+                  temporaryFilePath = path,
+                  originalPath = Paths.get(s"${job.title}/${parsedJob.metadata.map(meta => s"${meta.title}.${meta.extension}").getOrElse(job.id)}"),
+                  lastModifiedTime = None,
+                  manifest = manifest,
+                  esEvents = esEvents,
+                  ingestionServices = ingestionServices,
+                  annotations = annotations,
+                  fingerPrint = Some(fingerprint.toString)
+                ).process().asFuture,
+                15.minutes
+              )
+            }.toAttempt
+            _ <- remoteIngestStore.updateRemoteIngestJobStatus(parsedJob.id, "COMPLETED"/*, ingestFileResult.blob.uri TODO store blob uri on success */)
+          } yield job).fold(
+            failureDetail => {
+              val maybeParsedJob =  messageParseResult.asOpt
+              logger.error(s"Failed to ingest remote file for job with id ${maybeParsedJob.map(_.id).getOrElse(s"unknown (but had sqs id ${sqsMessage.getMessageId}")}: $failureDetail")
+              // TODO: Do we care if this sending to dead letter queue fails?
+              amazonSQSClient.sendMessage(config.outputDeadLetterQueueUrl, sqsMessage.getBody)
+              maybeParsedJob.map(parsedJob =>
+                remoteIngestStore.updateRemoteIngestJobStatus(parsedJob.id, "FAILED")
+              )
+            },
+            job => logger.info(s"Successfully ingested remote file for job with id ${job.id} in workspace ${job.workspaceId}")
+        ).onComplete { _ =>
+          amazonSQSClient.deleteMessage(config.outputQueueUrl, sqsMessage.getReceiptHandle)
+          //TODO: Delete file from S3
```

Comment:
this is probably crucial to fix before merging

Reply:
at this point we don't have the key for the file in s3, making it harder to delete. In the failure scenario we could have failed to fetch the item from neo4j, and we don't currently provide the key of the file in S3 in the message that comes back from SQS. There is a 7 day expiry on the bucket though as a fallback.

Reply:
(note we do delete the file in the success scenario)

Reply:
ahh great, shall we remove the TODO